transports/p2p/p2putil/addresses.go

```diff
@@ -0,0 +1,106 @@
+package p2putil
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/bitcoin-sv/block-headers-service/config"
+	"github.com/bitcoin-sv/block-headers-service/transports/p2p/addrmgr"
+)
+
+// Only setup a function to return new addresses to connect to when
+// not running in connect-only mode.  The simulation network is always
+// in connect-only mode since it is only intended to connect to
+// specified peers and actively avoid advertising and connecting to
+// discovered peers in order to prevent it from becoming a public test
+// network.
+func NewAddressFunc(getAddressFn func() *addrmgr.KnownAddress, outboundGroupCount func(string) int, lookupFn func(string) ([]net.IP, error)) func() (net.Addr, error) {
```

Comment:
Consider to define these functions as types:
```golang
type LookupFn func(string) ([]net.IP, error)
type GetAddressFn func() *addrmgr.KnownAddress
type CountOutboundGroupFn func(string) int
type FindNewAddrFn func() (net.Addr, error)

//...
func NewAddressFunc(getAddressFn GetAddressFn, outboundGroupCount CountOutboundGroupFn, lookupFn LookupFn) FindNewAddrFn {
```

Reply:
Done!